packages-exp/auth-exp/src/core/user/user_credential_impl.ts

```diff
@@ -46,4 +48,10 @@ export class UserCredentialImpl implements UserCredential {
     // updateAdditionalUserInfoFromIdTokenResponse(userCred, idTokenResponse);
     return userCred;
   }
+
+  static async _forOperation(user: User, operationType: externs.OperationType, response: PhoneOrOauthTokenResponse): Promise<UserCredentialImpl> {
```

Comment:
should we also use this for SIGN_IN?

Reply:
Let me clean this up in a separate PR. The `credential` param in `_fromIdTokenResponse` shouldn't be set as the `credential` field in the `UserCredential` object. Instead, it is just used to determine "isAnonymous". I will consolidate all of this into one method later